### DISCUSSION TITLE
"Sorry, this file is too large – the maximum size is 2 Bytes"

### DISCUSSION THREAD

**Discussion by Woitekku** [outline/outline#7593] [Self Hosting] [answered]
Unable to attach any file to the document. Despite of having FILE_STORAGE set to local or s3. 

```
Object { stack: "", … }
​
message: "Sorry, this file is too large – the maximum size is 2 Bytes"
​
name: "BadRequestError"
​
stack: "ExtendableError2@https://xxx.com/static/assets/index.OM0xKh-P.js:41:308623\nBadRequestError@https://xxx.com/static/assets/index.OM0xKh-P.js:41:309093\nApiClient/this.fetch@https://xxx.com/static/assets/index.OM0xKh-P.js:41:312439\n"
```

This is self-hosted installation on AWS EKS.
Ingress controller is based on AWS ALB.

AWS_S3_ACL=private
AWS_S3_FORCE_PATH_STYLE=false
AWS_S3_UPLOAD_BUCKET_NAME=xxx
AWS_S3_UPLOAD_BUCKET_URL=https://xxx.s3.eu-central-1.amazonaws.com
AWS_S3_UPLOAD_MAX_SIZE=2.62144e+07
AWS_WEB_IDENTITY_TOKEN_FILE=/var/run/secrets/eks.amazonaws.com/serviceaccount/token
FILE_STORAGE=s3
FILE_STORAGE_IMPORT_MAX_SIZE=2.62144e+07
FILE_STORAGE_LOCAL_ROOT_DIR=
FILE_STORAGE_UPLOAD_MAX_SIZE=2.62144e+07

**To Reproduce**
1. Create or edit any document
2. Try to put file into it

**Expected behavior**
Able to attach images to docs

**Screenshots**
If applicable, add screenshots or videos to help explain your problem.

**Outline (please complete the following information):**
- Install: [self hosted]
- Version: [outlinewiki/outline:latest]

**Desktop (please complete the following information):**
 - OS: [macOS Sonoma 14.4.1]
 - Browser [firefox]
 - Version [130]

**Comment by Jeffrey-F1solutions** [comment]
Hi,

Also getting this error saying the file is to big but its 2,8 MB
![image](https://github.com/user-attachments/assets/2122f71c-7024-4243-a019-f0e575d7d6f7)

File:
[CIS_Microsoft_365_Foundations_Benchmark_v3.1.0.pdf](https://github.com/user-attachments/files/16994219/CIS_Microsoft_365_Foundations_Benchmark_v3.1.0.pdf)


**Outline (please complete the following information):**
Install: [self hosted]
Version: [outlinewiki/outline:latest]

**Comment by tommoor** [comment, marked as answer]
Use an integer to define your file sizes, as in the documentation

```
FILE_STORAGE_IMPORT_MAX_SIZE=26214400
```

**Reply by Woitekku** [reply]
Oh! You are right! I did not notice that my helm chart push the value as string.

Was:
```
FILE_STORAGE_IMPORT_MAX_SIZE=2.62144e+07. 
AWS_S3_UPLOAD_MAX_SIZE=2.62144e+07. 
FILE_STORAGE_UPLOAD_MAX_SIZE=2.62144e+07
```

Is:
```
AWS_S3_UPLOAD_MAX_SIZE=26214400. 
FILE_STORAGE_UPLOAD_MAX_SIZE=26214400. 
FILE_STORAGE_IMPORT_MAX_SIZE=26214400
```

Now it works fine! Thanks.

**Comment by vember31** [comment]
It took me a while to figure this out so wanted to post it here too. I also am running Outline in kubernetes, using flux. 

Even though the values are expected to be integers, not strings, that means for the helmrelease, you _do_ need to make sure there are quotes, so that the number value doesn't get turned into scientific notation like in the post above. The quotes will guarantee that the value stays as-is. Outline interprets the string just fine it seems.

Thus, this fixed things for me and prevented the scientific notation from occurring:
```
              FILE_STORAGE: local
              FILE_STORAGE_UPLOAD_MAX_SIZE: "26214400"
              FILE_STORAGE_IMPORT_MAX_SIZE: "26214400"
```
